apis/cloud.redhat.com/v1alpha1/clowdapp_types.go

```diff
@@ -540,6 +544,26 @@ func (i *ClowdApp) GetClowdSAName() string {
 	return fmt.Sprintf("%s-app", i.GetClowdName())
 }
 
+func (i *ClowdApp) RunPreHook() bool {
+	// is there a better way to find out if the app has a job?
+	if i.Spec.PreHookJob.Name == "" || i.Annotations["clowder/pre-hook-status"] == "pending" {
```

Comment:
Specifically here I think that a comparison to `(i.Spec.PreHookJob{})` may do the job, but checking for the name is probably fine, as long as that is a mandatory field.